arm-notificationhubs/2016-03-01/swagger/notificationhubs.json

```diff
@@ -0,0 +1,1742 @@
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The name of the resource group."
+                }, {
+                    "name": "namespaceName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The namespace name."
+                }, {
+                    "name": "notificationHubName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The notification hub name."
+                }, {
+                    "$ref": "#/parameters/ApiVersionParameter"
+                }, {
+                    "$ref": "#/parameters/SubscriptionIdParameter"
+                }],
+                "responses": {
+                    "200": {
+                        "description": "",
+                        "schema": {
+                            "$ref": "#/definitions/NotificationHubResource"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "CheckAvailabilityParameters": {
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "Gets or sets name"
+                },
+                "location": {
+                    "type": "string",
+                    "description": "Gets or sets location."
+                },
+                "tags": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "string"
+                    },
+                    "description": "Gets or sets tags."
+                },
+                "isAvailiable": {
+                    "type": "boolean",
+                    "description": "Gets or sets true if the name is available and can be used to create new Namespace/NotificationHub. Otherwise false."
+                }
+            },
+            "required": [
+                "name"
+            ],
+            "description": "Parameters supplied to the Check Name Availability for Namespace and NotificationHubs."
+        },
+        "CheckAvailabilityResource": {
+            "properties": {
+                "id": {
+                    "type": "string",
+                    "description": "Gets or sets the id"
+                },
+                "location": {
+                    "type": "string",
+                    "description": "Gets or sets datacenter location"
+                },
+                "name": {
+                    "type": "string",
+                    "description": "Gets or sets name"
+                },
+                "type": {
+                    "type": "string",
+                    "description": "Gets or sets resource type"
+                },
+                "tags": {
```

Comment:
please make sure if all of these properties actually make sense.
Infact having a property named "reason" which explains the reason for the namespace not being available would make more sense then the set of properties documented here.